packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts

```diff
@@ -162,12 +162,19 @@ export class OdspDriverUrlResolver implements IUrlResolver {
         resolvedUrl: IResolvedUrl,
         relativeUrl: string,
         packageInfoSource?: IContainerPackageInfo,
+        redirectLocation?: string,
     ): Promise<string> {
         let dataStorePath = relativeUrl;
         if (dataStorePath.startsWith("/")) {
             dataStorePath = dataStorePath.substr(1);
         }
         const odspResolvedUrl = getOdspResolvedUrl(resolvedUrl);
+        if (redirectLocation !== undefined) {
+            // Generate the new SiteUrl from the redirection location.
+            const newSiteDomain = new URL(redirectLocation).origin;
+            const newSiteUrl = `${newSiteDomain}${new URL(odspResolvedUrl.siteUrl).pathname}`;
```

Comment:
Correct me if I got it wrong, but I believe we (ideally) can deal with domain rename, but also with site move (within same domain). Glancing at this code, I feel like it deals only with the former.